pkg/receive/handler.go

```diff
@@ -258,6 +258,8 @@ type replica struct {
 func (h *Handler) handleRequest(ctx context.Context, rep uint64, tenant string, wreq *prompb.WriteRequest) error {
 	// The replica value in the header is one-indexed, thus we need >.
 	if rep > h.options.ReplicationFactor {
+		level.Error(h.logger).Log("err", errBadReplica, "msg", "write request rejected",
```

Comment:
Can we do this in the place where error is handled? Handling error twice (by returning and logging) is a bad smell 🙈

Reply:
We could :+1:  I chose not to as we would then need to replicate this in two places in a switch statement ([here](https://github.com/thanos-io/thanos/blob/f1597d785b7f33dc3912faa6b993198bd68eae1b/pkg/receive/handler.go#L352) & [here](https://github.com/thanos-io/thanos/blob/f1597d785b7f33dc3912faa6b993198bd68eae1b/pkg/receive/handler.go#L656)).

I felt this was slightly tidier, but happy to be convinced otherwise :man_shrugging:

Reply:
Hm... there is a risk we will handle things twice, but not a biggie! (: